### PR TITLE
urxvt: add module

### PR DESCRIPTION
**Pull request by uvNikita**
Will be happy to get a feedback on these options specifically:

- scroll.keepPostion
- scroll.downOnInput
- scroll.downOnOutput

Are they understandable, or maybe it's better to keep urxvt names?

### CODE REVIEW

File: modules/programs/urxvt.nix [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+    fonts = mkOption {
+      type = types.listOf types.str;
+      default = [];
+      description = "List of fonts to be used.";
+      example = [ "xft:Droid Sans Mono Nerd Font:size=9" ];
+    };
+
+    keybindings = mkOption {
+      type = types.attrsOf types.string;
```
Comment (rycee) [inline modules/programs/urxvt.nix]: Should be `types.str`.

File: modules/programs/urxvt.nix [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+              default = true;
+              description = "Whether to display an rxvt scrollbar without a trough.";
+            };
+          };
+        };
+        description = "Scrollbar settings.";
+      };
+
+      lines = mkOption {
+        type = types.int;
```
Comment (rycee) [inline modules/programs/urxvt.nix]: Maybe `types.ints.unsigned`?

File: modules/programs/urxvt.nix [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+      keepPosition = mkOption {
+        type = types.bool;
+        default = true;
+        description = "Whether to keep a scroll position when tty receives new lines.";
+      };
+
+      downOnInput = mkOption {
+        type = types.bool;
+        default = true;
+        description = "Whether to scroll to bottom on keboard input.";
```
Comment (rycee) [inline modules/programs/urxvt.nix]: s/keboard/keyboard/

File: modules/programs/urxvt.nix [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+      lines = mkOption {
+        type = types.int;
+        default = 64;
+        description = "Number of lines to save in the scrollback buffer.";
+      };
+
+      keepPosition = mkOption {
+        type = types.bool;
+        default = true;
+        description = "Whether to keep a scroll position when tty receives new lines.";
```
Comment (rycee) [inline modules/programs/urxvt.nix]: s/tty/TTY/

File: modules/programs/urxvt.nix [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+    };
+
+    iso14755 = mkOption {
+      type = types.bool;
+      default = true;
+      description = "ISO14755 support for viewing and entering unicode characters.";
+    };
+
+    scroll = {
+      bar = mkOption {
```
Comment (rycee) [inline modules/programs/urxvt.nix]: Should the `scroll.bar` option have a default of `{}`?

File: modules/programs/urxvt.nix [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+        description = "Number of lines to save in the scrollback buffer.";
+      };
+
+      keepPosition = mkOption {
+        type = types.bool;
+        default = true;
+        description = "Whether to keep a scroll position when tty receives new lines.";
+      };
+
+      downOnInput = mkOption {
```
Comment (rycee) [inline modules/programs/urxvt.nix]: I'd suggest `scrollOnKeystroke` to match `programs.termite.scrollOnKeystroke`.

File: modules/programs/urxvt.nix [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+        type = types.bool;
+        default = true;
+        description = "Whether to scroll to bottom on keboard input.";
+      };
+
+      downOnOutput = mkOption {
```
Comment (rycee) [inline modules/programs/urxvt.nix]: I'd suggest `scrollOnOutput` to match `programs.gnome-terminal.profile.<name>.scrollOnOutput` and `programs.termite.scrollOnOutput`.

Yeah, it's a bit unfortunate to have `scroll.scrollOnOutput` but perhaps nice to have consistent naming here 🙂

File: modules/programs/urxvt.nix [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+      description = "Whether to enable pseudo-transparency.";
+    };
+
+    shading = mkOption {
+      type = types.int;
+      default = 100;
+      description = "Darken (0 .. 99) or lighten (101 .. 200) the transparent background.";
+    };
+
+    extraConfig = mkOption {
```
Comment (rycee) [inline modules/programs/urxvt.nix]: Not used?

File: modules/programs/urxvt.nix [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+        description = "Scrollbar settings.";
+      };
+
+      lines = mkOption {
+        type = types.int;
+        default = 64;
+        description = "Number of lines to save in the scrollback buffer.";
+      };
+
+      keepPosition = mkOption {
```
Comment (rycee) [inline modules/programs/urxvt.nix]: Seems like a good option name to me.

File: modules/programs/urxvt.nix [hunk 10]
```diff
@@ -0,0 +1,150 @@
...
+              description = "Whether to display an rxvt scrollbar without a trough.";
+            };
+          };
+        };
+        description = "Scrollbar settings.";
+      };
+
+      lines = mkOption {
+        type = types.int;
+        default = 64;
```
Comment (rycee) [inline modules/programs/urxvt.nix]: This seems to be very little. The gnome-terminal and termite modules use 10000 as the default value, perhaps could reuse that here?

File: modules/programs/urxvt.nix (hunk 7 above)
Comment (uvNikita) [reply modules/programs/urxvt.nix]: Great suggestion. Agree to both `scrollOnOutput` and `scrollOnKeystroke`, nice to be consistent.

File: modules/programs/urxvt.nix (hunk 10 above)
Comment (uvNikita) [reply modules/programs/urxvt.nix]: Yeah, I definitely agree. This was a default value in urxvt, but I also think we should increase it.

**Comment by rycee** [comment]
Yep, looking good! 🙂

Rebased to master in c17f37857cdb134f8ffbf5e2ee8c8ab317136e0f. I updated the news timestamp and changed the `programs.urxvt.keybindings` option to use `literalExample` to prettify the documentation.

**Pull request closed without merging**